tests/unit/src/test/scala/tests/debug/BreakpointDapSuite.scala

```diff
@@ -0,0 +1,468 @@
+package tests.debug
+import tests.BaseDapSuite
+import scala.meta.internal.metals.debug.DebugStep._
+import scala.meta.internal.metals.debug.DebugFileLayout
+import scala.meta.internal.metals.debug.StepNavigator
+
+object BreakpointDapSuite extends BaseDapSuite("debug-breakpoint") {
+
+  // disabled, because finding enclosing class for the breakpoint line is not working
+  // see [[scala.meta.internal.metals.debug.SetBreakpointsRequestHandler]]
+  assertFileBreakpoints("preceding-class", disabled = true)(
```

Comment:
Interesting, this should be totally fixable inside `Mtags.toplevels`. I've considered extending the parser to emit global def/val/var symbols as well. Can we open a ticket for this?

Reply:
This is BTW a totally fine limitation for now, we'll make it clear in the release announcement that the debugging support has known issues

Reply:
Opened https://github.com/scalameta/metals/issues/1195